pkgs/development/python-modules/pytest-docker/default.nix

```diff
@@ -0,0 +1,60 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+  wheel,
+  attrs,
+  pytest,
+  docker-compose,
+  mypy,
+  pytest-mypy,
+  pycodestyle,
+  pytest-pylint,
+  requests,
+  types-requests,
+  types-setuptools,
+}:
+
+buildPythonPackage rec {
+  pname = "pytest-docker";
+  version = "3.1.1";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-I3FSSASnUqqnZsebnu6OY0U0r924JZfztXPafF1v+18=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    attrs
+    pytest
+  ];
+
+  passthru.optional-dependencies = {
+    docker-compose-v1 = [ docker-compose ];
+    tests = [
+      mypy
+      pytest-mypy
+      pycodestyle
+      pytest-pylint
+      requests
+      types-requests
+      types-setuptools
+    ];
+  };
+
+  pythonImportsCheck = [ "pytest_docker" ];
+
```

Comment:
Could you enable tests with `pytestCheckHook`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook